### PR TITLE
fix(tianm):oltp_read_write.lua sysbench returned error 1366(#1371)

### LINKED ISSUE #1371: bug: oltp_read_write.lua sysbench returned error 1366 (Incorrect string value..）
### Have you read the Contributing Guidelines on issues?

- [X] I have read the [Contributing Guidelines on issues](https://github.com/stoneatom/stonedb/blob/stonedb-5.7-dev/CONTRIBUTING.md).

### Please confirm if bug report does NOT exists already ?

- [X] I confirm there is no existing issue for this

### Describe the problem

```shell
[root@HAMI02 log]# sysbench --mysql-port=3306 --mysql-user=root --mysql-password=123456 --mysql-socket=/opt/stonedb57/install//tmp/mysql.sock --mysql-db=zz  --mysql_storage_engine=tianmu --create_secondary=off --table_size=20240 --tables=5 --events=0 --time=60 --threads=10 --percentile=95 --report-interval=5 --auto_inc=true /usr/share/sysbench/oltp_read_write.lua run
sysbench 1.0.17 (using system LuaJIT 2.0.4)

Running the test with following options:
Number of threads: 10
Report intermediate results every 5 second(s)
Initializing random number generator from current time


Initializing worker threads...

Threads started!

FATAL: mysql_stmt_execute() returned error 1366 (Incorrect string value: '\xE7'\x00\x00\x00\x00...' for column 'c' at row 1) for query 'DELETE FROM sbtest5 WHERE id=?'
FATAL: `thread_run' function failed: /usr/share/sysbench/oltp_common.lua:487: SQL error, errno = 1366, state = 'HY000': Incorrect string value: '\xE7'\x00\x00\x00\x00...' for column 'c' at row 1
FATAL: mysql_stmt_execute() returned error 1366 (Incorrect string value: '\xE7'\x00\x00\x00\x00...' for column 'c' at row 4) for query 'UPDATE sbtest5 SET c=? WHERE id=?'
FATAL: `thread_run' function failed: /usr/share/sysbench/oltp_common.lua:469: SQL error, errno = 1366, state = 'HY000': Incorrect string value: '\xE7'\x00\x00\x00\x00...' for column 'c' at row 4
FATAL: mysql_stmt_execute() returned error 1366 (Incorrect string value: '\xE7'\x00\x00\x00\x00...' for column 'c' at row 4) for query 'UPDATE sbtest5 SET k=k+1 WHERE id=?'
FATAL: `thread_run' function failed: /usr/share/sysbench/oltp_common.lua:458: SQL error, errno = 1366, state = 'HY000': Incorrect string value: '\xE7'\x00\x00\x00\x00...' for column 'c' at row 4

```

### Expected behavior

_No response_

### How To Reproduce

_No response_

### Environment

```shell
[root@HAMI02 ~]# /opt/stonedb57/install//bin/mysqld --version
/opt/stonedb57/install//bin/mysqld  Ver 5.7.36-StoneDB-v1.0.2 for Linux on x86_64 (build-)
build information as follow:
        Repository address: https://github.com/stoneatom/stonedb.git:HEAD
        Branch name: HEAD
        Last commit ID: e8f079440
        Last commit time: Date:   Wed Mar 8 19:10:15 2023 +0800
        Build time: Date: Wed Mar  8 13:06:25 UTC 2023
```

### Are you interested in submitting a PR to solve the problem?

- [ ] Yes, I will!

### PR DESCRIPTION
**Pull request by konghaiya**
Cause of the problem:
The reason for the problem is that when the update and update data are merged, there is a place where the update bitmap is not set, resulting in the subsequent data reading confusion

## Summary about this PR

Issue Number: close #1371

## Tests Check List

- [ ] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

## Changelog

- [ ] New Feature
- [x] Bug Fix
- [ ] Performance Improvement
- [ ] Build/Testing/CI/CD
- [ ] Documentation
- [ ] Not for changelog (changelog entry is not required)

## Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features

### CODE REVIEW

File: storage/tianmu/util/bitset.h [hunk 1]
```diff
@@ -32,6 +33,7 @@ class BitSet {
       ptr_ = new char[(sz + NO_OF_BITS - 1) / NO_OF_BITS]();
       allocated_ = true;
     }
+    memset(ptr_, 0, (sz + NO_OF_BITS - 1) / NO_OF_BITS);
```
Comment (RingsC) [inline storage/tianmu/util/bitset.h]: You will reset the `data`, which it maybe already stored some data in.

**Review by RingsC — CHANGES_REQUESTED** [review]

File: storage/tianmu/util/bitset.h (hunk 1 above)
Comment (konghaiya) [reply storage/tianmu/util/bitset.h]: Yes, it is. I think how to change it

Comment (RingsC) [reply storage/tianmu/util/bitset.h]: > Yes, it is. I think how to change it

What's your solution?

Comment (konghaiya) [reply storage/tianmu/util/bitset.h]: The reason for the problem is that when the update and update data are merged, there is a place where the update bitmap is not set, resulting in the subsequent data reading confusion

**Review by RingsC — APPROVED** [review]
LGTM

**Review by duanfuxiang0 — APPROVED** [review]
LGTM

**Pull request merged**